### PR TITLE
Added Spain Balearic Islands parsers

### PR DESCRIPTION
**Pull request by blackleg**

### CODE REVIEW

File: parsers/config.py [hunk 1]
```diff
@@ -165,6 +168,7 @@
     # ES
     'ES->FR':     ENTSOE.fetch_exchange,
     'ES->PT':     ENTSOE.fetch_exchange,
+    'ES->ES-IB':  ES_IB.fetch_exchange,
```
Comment (corradio) [inline parsers/config.py]: Cool!

File: parsers/ES_IB.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+# The request library is used to fetch content through HTTP
+from requests import Session
+from reescraper import BalearicIslands
+
+
+def fetch_consumption(country_code='ES-IB', session=None):
+    ses = session or session()
+    response = BalearicIslands(ses).get()
+
+    if not response:
```
Comment (corradio) [inline parsers/ES_IB.py]: If you don't get a response you should abort early and return `None`

File: parsers/ES_IB.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    if response:
+        data['production']['coal'] = response.carbon
+        data['production']['gas'] = response.gas + response.combined
+        data['production']['solar'] = response.solar
+        data['production']['oil'] = response.vapor + response.diesel
+        data['production']['wind'] = response.wind
+
+        hidro = response.hydraulic
```
Comment (corradio) [inline parsers/ES_IB.py]: Is it possible de separate pumped hydro from river?
If this is only pumped hydro, then production from pumped hydro should be counted as negative storage and not positive production (like drawing power from a battery)

File: parsers/ES_IB.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        data['production']['unknown'] = response.unknown()
+
+    return data
+
+
+def fetch_exchange(country_code1='ES', country_code2='ES-IB', session=None):
+
+    ses = session or session()
+    response = BalearicIslands(ses).get()
+    if not response:
```
Comment (corradio) [inline parsers/ES_IB.py]: Same here

File: parsers/ES_IB.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+    return data
+
+
+def fetch_production(country_code='ES-IB', session=None):
+
+    ses = session or session()
+    response = BalearicIslands(ses).get()
+
+    if not response:
```
Comment (corradio) [inline parsers/ES_IB.py]: Same here

File: parsers/ES_IB.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+    data = {
+        'sortedCountryCodes': '->'.join(sorted_country_codes),
+        'datetime': datetime,
+        'source': 'demanda.ree.es',
+    }
+
+    if response:
+        netFlow = response.link['pe_ma']
+        data['netFlow'] = netFlow if country_code1 == sorted_country_codes[0] else -1 * netFlow
```
Comment (corradio) [inline parsers/ES_IB.py]: do you confirm that `response.link['pe_ma']` is flow from ES to ES-IB, i.e. a positive value represents the ES-IB importing power?

**Comment by corradio** [comment]
Thanks for adding this!
Forgot to ask: can you also update the README with the data source?
Also would be nice to know if this island has pumped hydro and/or run-of-the-river, and if we can separate them.

**Comment by blackleg** [comment]
@corradio The Balearic Islands electric system don't have hydraulic generation systems.

I maintained the code to read the hydraulic production because I reused the code from the Canary Islands parser.

File: parsers/ES_IB.py (hunk 6 above)
Comment (blackleg) [inline parsers/ES_IB.py]: Positive value represents ES-IB is importing power, negative value is exporting to ES

**Pull request merged**